R/notin.R

```diff
@@ -0,0 +1,9 @@
+"%notin%" = function(x, table) {
+  if (is.character(x) && is.character((table))) {
+    result = .Call(Cchin, x, table)
+  } else {
+    result = match(x, table, nomatch = 0) > 0
+  }
+  return(.Call(Cnegate, result))
```

Comment:
Bit uncomfortable with a `.Call` doing this by reference. I suppose in this simple function as it stands it will work but there is nothing to indicate this changes the logical vector by reference to the reader of the code. But I see what you mean about reusing it to negate the output from `match`/`%in%` too which is nice and better than what I had in mind which was `Cchnotin` wrapper just for that branch. Ok so maybe naming it `CnegateByRef` would be clear so that a comment isn't as critical to remember in future if we use it elsewhere too.

Reply:
I agree that it doesn't seem great to be altering vectors by reference from R, so I've moved the negation to C where it can be used more intuitively.